crates/project/src/task_inventory.rs

```diff
@@ -294,6 +294,7 @@ impl Inventory {
                         let (previous_task, _) = o.get();
                         let new_template = task.original_task();
                         if new_template.ignore_previously_resolved
+                            || new_template.reevaluate_context
```

Comment:
I think we should merge this and `ignore_previously_resolved` one together: the latter was added for `"run selection"`-labeled tasks that always have the same label, but have to run the new command every time it's spawned.

Currently, those are rerun with the same arguments and the PR will change it if merges the two knobs together, but this seems fine, as the new PR provides a way to force context reuse with the `reevaluate_context: Option<bool>,` change in the action.

Looks like the new properly better have the `reevaluate_context` and unite the behavior of both properties from this PR.

Reply:
One nasty side effect of this removal is the fact that the reevaluated task is "never used" and will always be sorted down below all history in the task modal, which is somewhat confusing.

The problem is that if we leave it as is, old resolved entry will have wrong substitutions in the label/hover/etc. , so we need to resolve the template with the new context (we should be doing the resolutions above already?) and insert it with the old lru score.